pkg/cpi/mockingUtils.go

```diff
@@ -433,9 +469,128 @@ func GetFunctionNameByTestTypeAndMethod(method, testType string) string {
+
+	return GetMockResponseByTestTypeAndMockFunctionName("GetIntegrationArtifactDeployStatusMockResponse", testType)
+}
+
+//GetIntegrationArtifactDeployErrorDetailsMockResponse -Provide http respose body
+func GetIntegrationArtifactDeployErrorDetailsMockResponse(testType string) (*http.Response, error) {
+
+	return GetMockResponseByTestTypeAndMockFunctionName("GetIntegrationArtifactDeployErrorDetailsMockResponse", "PostiveAndGetDeployedIntegrationDesigntimeArtifactErrorResBody")
+}
+
+//GetIntegrationArtifactDeployStatusMockResponseBody -Provide http respose body
+func GetIntegrationArtifactDeployStatusMockResponseBody() (*http.Response, error) {
+
+	resp := http.Response{
+		StatusCode: 200,
+		Body: ioutil.NopCloser(bytes.NewReader([]byte(`{
+			"d": {
+				"__metadata": {
+					"id": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('flow1')",
+					"uri": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('flow1')",
+					"media_src": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('flow1')/$value",
+					"edit_media": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('flow1')/$value"
+				},
+				"Id": "flow1",
+				"Version": "1.0.24",
+				"Name": "flow1",
+				"Status": "STARTED",
+				"ErrorInformation": {
+					"__deferred": {
+						"uri": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('flow1')/ErrorInformation"
+					}
+				}
+			}
+		}`))),
+	}
+	return &resp, nil
+}
+
+//GetIntegrationArtifactDeployStatusErrorMockResponseBody -Provide http respose body
+func GetIntegrationArtifactDeployStatusErrorMockResponseBody() (*http.Response, error) {
+
+	resp := http.Response{
+		StatusCode: 200,
+		Body: ioutil.NopCloser(bytes.NewReader([]byte(`{
+			"d": {
+				"__metadata": {
+					"id": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('smtp')",
+					"uri": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('smtp')",
+					"media_src": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('smtp')/$value",
+					"edit_media": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('smtp')/$value"
+				},
+				"Id": "smtp",
+				"Version": "2.0",
+				"Name": "smtp",
+				"Status": "ERROR",
+				"ErrorInformation": {
+					"__deferred": {
+						"uri": "https://roverpoc.it-accd002.cfapps.sap.hana.ondemand.com/api/v1/IntegrationRuntimeArtifacts('smtp')/ErrorInformation"
+					}
+				}
+			}
+		}`))),
+	}
+	return &resp, nil
+}
+
+//GetIntegrationArtifactDeployErrorStatusMockResponseBody -Provide http respose body
+func GetIntegrationArtifactDeployErrorStatusMockResponseBody() (*http.Response, error) {
+
+	resp := http.Response{
+		StatusCode: 200,
```

Comment:
Unrelated, but: You get status code 200 in case of an error?

Reply:
this mock function is to provide error details in case of error info API succeeds. for Error info API to succeed we need to give 200 status code

Reply:
Ah, so it is a GET request to retrieve the error details, and this GET should return 200 if it is successful.